src/ldp.rs

```diff
@@ -102,12 +216,7 @@ pub trait ProofSuite {
         proof: &Proof,
         document: &(dyn LinkedDataDocument + Sync),
         resolver: &dyn DIDResolver,
-    ) -> Result<(), Error>
-    where
-        Self: Sized,
-    {
-        verify(proof, document, resolver).await
-    }
```

Comment:
Why the removal default implementation (to avoid issues due to the implicit behaviour I assume)?

I can't remember if I asked already but why do we have this chain of calls? And if `verify` was a trait of the proof, we could have a blanket implementation (but that's assuming the behaviour is always the same)?

Reply:
With the default implementation (f6c3cd2a373dd06f1464f23a56bae458ce4c3189):
```

error: the `verify` method cannot be invoked on a trait object
   --> src/ldp.rs:329:15
    |
221 |         Self: Sized,
    |               ----- this has a `Sized` requirement
...
329 |         suite.verify(proof, document, resolver).await
    |               ^^^^^^
```

Removing the Sized requirement (604623203adf3d5ef605f8bf4ea506f1bb30549e) works but says will become a hard error in a future release:
```
warning: the trait `ProofSuite` cannot be made into an object
   --> src/ldp.rs:214:14
    |
214 |     async fn verify(
    |              ^^^^^^
    |
    = note: `#[warn(where_clauses_object_safety)]` on by default
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #51443 <https://github.com/rust-lang/rust/issues/51443>
note: for a trait to be "object safe" it needs to allow building a vtable to allow the call to be resolvable dynamically; for more information visi
t <https://doc.rust-lang.org/reference/items/traits.html#object-safety>
   --> src/ldp.rs:214:14
    |
191 | pub trait ProofSuite {
    |           ---------- this trait cannot be made into an object...
...
214 |     async fn verify(
    |              ^^^^^^ ...because method `verify` references the `Self` type in its `where` clause
    = help: consider moving `verify` to another trait
```

Reply:
I hesitate to apply the above changes (I pushed them to a temporary branch, because of the "it will become a hard error in a future release" warning.

> why do we have this chain of calls?

The verify function called in the (removed) default implementation is for verifying a linked data signature using detached JWS over the hash of the canonicalized proof options and linked data document. This is a common case, used in e.g. Ed25519Signature2018 and JsonWebSignature2020. But some proof/signature suites don't use this.

Reply:
Oh I'm sorry I completely misread your first comment. IMO I think it shows that `ProofSuite` shouldn't exist, and instead `Proof` should implement those methods. Then you would have a `match` on the `type_` and for signatures that use JWS it would route to `jws::verify`. Which would mean less code duplication, less monomorphisation and a code flow that's maybe easier to follow? But I might just be nitpicking. I leave it to your good judgement.